File: src/compiler/Token.cpp
#include "Token.h"

Token::Token() : type(TokenType::TOKEN_UNKNOWN), lexeme(""), line(0), col(0) {}

Token::Token(TokenType type, const std::string& lexeme, int line, int col)
    : type(type), lexeme(lexeme), line(line), col(col) {}

std::string Token::toString() const {
    return "[" + tokenTypeToString(type) + " \"" + lexeme + "\" L:" +
           std::to_string(line) + " C:" + std::to_string(col) + "]";
}

File: dictatorscript-vscode/shared/src/token.ts
/**
 * DictatorScript Token
 *
 * Represents a single token produced by the lexer.
 * Positions are 0-based (LSP convention).
 */

import { TokenType } from './tokenType';

export interface Token {
    /** The token classification. */
    type: TokenType;
    /** The raw source text of the token. */
    lexeme: string;
    /** 0-based line number. */
    line: number;
    /** 0-based column number. */
    col: number;
    /** Absolute character offset from the start of the source. */
    offset: number;
    /** Length of the token in characters. */
    length: number;
}
